aggregate_root/lib/aggregate_root/instrumented_repository.rb

```diff
@@ -0,0 +1,33 @@
+module AggregateRoot
+  class InstrumentedRepository
+    def initialize(repository, instrumentation)
+      @repository = repository
+      @instrumentation = instrumentation
+    end
+
+    def load(aggregate, stream_name)
+      instrumentation.instrument("load.repository.aggregate_root",
+                                 aggregate_class: aggregate.class,
+                                 stream_name: stream_name) do
+        repository.load(aggregate, stream_name)
+      end
+    end
+
+    def store(aggregate, stream_name)
+      instrumentation.instrument("store.repository.aggregate_root",
+                                 aggregate_class: aggregate.class,
```

Comment:
I think passing the instance, labeled as aggregate would be fine (thus s/aggregate_class/aggregate/)

Reply:
Instrumentation is useful in logging context, being able to track particular instance is helping.